frontend/src/lib/utils/eventUsageLogic.ts

```diff
@@ -0,0 +1,64 @@
+/* This file contains the logic to report custom frontend events */
+import { kea } from 'kea'
+import { keyMapping } from 'lib/components/PropertyKeyInfo'
+import posthog from 'posthog-js'
+import { userLogic } from 'scenes/userLogic'
+import { eventUsageLogicType } from 'types/lib/utils/eventUsageLogicType'
+import { AnnotationType } from '~/types'
+
+const keyMappingKeys = Object.keys(keyMapping.event)
+
+export const eventUsageLogic = kea<eventUsageLogicType>({
+    actions: {
+        reportAnnotationViewed: (annotations) => ({ annotations }),
+        reportPersonDetailViewed: (person) => ({ person }),
+    },
+    listeners: {
+        reportAnnotationViewed: async ({ annotations }: { annotations: AnnotationType[] | null }, breakpoint) => {
+            if (!annotations) {
```

Comment:
Did not test this, but I'd assume breakpoint(500) would need to be first right - otherwise we'd still report it as viewed.

Reply:
Q: Is it worth having code for the cancel case at all - KISS.

Reply:
Don't think the breakpoint should go first, because the `annotations` won't change while the breakpoint is happening. If the variable is null, why wait, it will never lead to something getting reported anyways.

I clarified the comment a bit. Basically if the var is not set, we can't report usage.